libcaf_test/caf/test/outline.cpp

```diff
@@ -107,7 +107,7 @@ void outline::run() {
     }
   }
   ctx_->parameters = ctx_->example_parameters[ctx_->example_id];
-  auto guard = detail::make_scope_guard([this] {
+  auto guard = detail::make_scope_guard([this]() noexcept {
     auto& ptr = ctx_->steps[std::make_pair(0, ctx_->example_id)];
```

Comment:
This can technically throw, so clang-tidy might complain later about it. We can just move it out of the lambda and capture it instead.